### PR TITLE
Added cpu and memory utilization details.

### PR DESCRIPTION
**Pull request by nnDarshan**
This patch adds cpu and memory utilization details
to cpu and memory objects, with this change these
details will be synced by node-agent after every
node sync interval

tendrl-bug-id : Tendrl/node-agent#585
Signed-off-by: nnDarshan <darshan.n.2024@gmail.com>

### CODE REVIEW

**Comment by nnDarshan** [comment]
@shtripat @r0h4n @brainfunked @nthomas-redhat Pls review

**Comment by nnDarshan** [comment]
@anivargi Please have a look. Wanted to make sure this change does not break API

File: tendrl/commons/objects/cpu/__init__.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import psutil
 from tendrl.commons import objects
```
Comment (r0h4n) [inline tendrl/commons/objects/cpu/__init__.py]: you need to separate imports by 2 blank lines
- python stdlibs
- third party (like psutil)
- imports from tendrl

File: tendrl/commons/objects/memory/__init__.py [hunk 2]
```diff
@@ -1,56 +1,23 @@
-from tendrl.commons.event import Event
-from tendrl.commons.message import ExceptionMessage
+import psutil
```
Comment (r0h4n) [inline tendrl/commons/objects/memory/__init__.py]: you need to separate imports by 2 blank lines
- python stdlibs
- third party (like psutil)
- imports from tendrl

File: tendrl/commons/objects/memory/__init__.py [hunk 3]
```diff
@@ -1,56 +1,23 @@
...
+                 swap_available=None, swap_percent=None,
                  *args, **kwargs):
         super(Memory, self).__init__(*args, **kwargs)
-        memory = self._getNodeMemory()
-        self.total_size = total_size or memory["TotalSize"]
-        self.swap_total = swap_total or memory["SwapTotal"]
+        mem = psutil.virtual_memory()
+        self.memory_size = memory_size or str(mem.total)
+        self.memory_available = memory_available or str(mem.available)
+        self.memory_percent = memory_percent or str(mem.percent)
```
Comment (r0h4n) [inline tendrl/commons/objects/memory/__init__.py]: if this is memory used percent, call it "memory_util_percent"

File: tendrl/commons/objects/memory/__init__.py [hunk 4]
```diff
@@ -1,56 +1,23 @@
...
+    def __init__(self, memory_size=None, memory_available=None,
+                 memory_percent=None, swap_size=None,
+                 swap_available=None, swap_percent=None,
                  *args, **kwargs):
         super(Memory, self).__init__(*args, **kwargs)
-        memory = self._getNodeMemory()
-        self.total_size = total_size or memory["TotalSize"]
-        self.swap_total = swap_total or memory["SwapTotal"]
+        mem = psutil.virtual_memory()
+        self.memory_size = memory_size or str(mem.total)
```
Comment (r0h4n) [inline tendrl/commons/objects/memory/__init__.py]: call this "memory_total"

File: tendrl/commons/objects/cpu/__init__.py (hunk 1 above)
Comment (nnDarshan) [reply tendrl/commons/objects/cpu/__init__.py]: ack

File: tendrl/commons/objects/memory/__init__.py (hunk 2 above)
Comment (nnDarshan) [reply tendrl/commons/objects/memory/__init__.py]: ack

File: tendrl/commons/objects/memory/__init__.py (hunk 4 above)
Comment (nnDarshan) [reply tendrl/commons/objects/memory/__init__.py]: ack

File: tendrl/commons/objects/memory/__init__.py (hunk 3 above)
Comment (nnDarshan) [reply tendrl/commons/objects/memory/__init__.py]: ack

**Comment by r0h4n** [comment]
@Tendrl/tendrl-core please review

**Pull request merged**